operator/src/test/java/org/bf2/operator/managers/StrimziManagerTest.java

```diff
@@ -205,6 +214,30 @@ public void testUninstallingStrimziOperator() {
         assertFalse(checkStrimziVersion(strimziVersions, "strimzi-cluster-operator.v2", true));
     }
 
+    @Test
+    public void testStrimziLeadershipLeaseCleanedOnDelete() {
+        final Lease lease = new LeaseBuilder()
+                .withNewMetadata()
+                    .withName("strimzi-cluster-operator.v1-leadership-token")
+                    .withNamespace("ns-1")
+                .endMetadata()
+                .withNewSpec()
+                    .withHolderIdentity("me")
+                    .withLeaseDurationSeconds(1000)
+                    .withAcquireTime(ZonedDateTime.ofInstant(Instant.now(), ZoneId.of("UTC")))
+                .endSpec().build();
+        server.getClient().leases().inNamespace("ns-1").create(lease);
+        final Deployment deployment = installStrimziOperator("strimzi-cluster-operator.v1", "ns-1", "2.7.0=kafka-2.7.0", true, true);
+        final Resource<Lease> leaseResource = server.getClient().leases().inNamespace("ns-1").withName("strimzi-cluster-operator.v1-leadership-token");
+        Lease actual = leaseResource.get();
+        assertNotNull(actual);
+        List<StrimziVersionStatus> strimziVersions = this.strimziManager.getStrimziVersions();
+        assertTrue(checkStrimziVersion(strimziVersions, "strimzi-cluster-operator.v1", true));
+        strimziManager.deleteLeadershipLeaseIfPresent(deployment);
```

Comment:
I wasn't sure how to invoke the ResourceEventHandler which the test passes into a `MockResourceInformerFactory`, maybe with some re-working we could keep a reference somewhere in the mock. Maybe a future improvement to make them testable.

Reply:
I went and had a shot at this, keeping references to all handlers created per type so that the test can access and call them directly. Maybe it would be better if `StrimziManager` implemented `ResourceEventHandler<Deployment>` and we didn't us an anonymous inner class, then we could call the methods directly on StrimziManager.